src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItemsFactory.cs

```diff
@@ -0,0 +1,170 @@
+﻿// MUX Reference NavigationViewItemPresenter.cpp, commit 5e503fa
+
+using System;
+using System.Collections.Generic;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	internal class NavigationViewItemsFactory : ElementFactory
+	{
+		private IElementFactoryShim m_itemTemplateWrapper = null;
+		private NavigationViewItemBase m_settingsItem = null;
+		private List<NavigationViewItem> navigationViewItemPool;
+
+		internal void UserElementFactory(object newValue)
+		{
+			m_itemTemplateWrapper = newValue as IElementFactoryShim;
+			if (m_itemTemplateWrapper == null)
+			{
+				// ItemTemplate set does not implement IElementFactoryShim. We also 
+				// want to support DataTemplate and DataTemplateSelectors automagically.
+				if (newValue is DataTemplate dataTemplate)
+				{
+					m_itemTemplateWrapper = new ItemTemplateWrapper(dataTemplate);
+				}
+				else if (newValue is DataTemplateSelector selector)
+				{
+					m_itemTemplateWrapper = new ItemTemplateWrapper(selector);
+				}
+			}
+
+			navigationViewItemPool = new List<NavigationViewItem>();
+		}
+
+		internal void SettingsItem(NavigationViewItemBase settingsItem)
+		{
+			m_settingsItem = settingsItem;
+		}
+
+		// Retrieve the element that will be displayed for a specific data item.
+		// If the resolved element is not derived from NavigationViewItemBase, wrap in a NavigationViewItem before returning.
+		protected override UIElement GetElementCore(ElementFactoryGetArgs args)
+		{
+			object GetNewContent(IElementFactoryShim itemTemplateWrapper, NavigationViewItemBase settingsItem)
+			{
+				// Do not template SettingsItem
+				if (settingsItem != null && settingsItem == args.Data)
+				{
+					return args.Data;
+				}
+
+				if (itemTemplateWrapper != null)
+				{
+					return itemTemplateWrapper.GetElement(args);
+				}
+				return args.Data;
+			}
+
+			var newContent = GetNewContent(m_itemTemplateWrapper, m_settingsItem);
+
+			// Element is already of expected type, just return it
+			if (newContent is NavigationViewItemBase newItem)
+			{
+				return newItem;
+			}
+
+			// Accidentally adding a OS XAML NavigationViewItem to WinUI's NavigationView can cause unnecessary confusion for developers
+			// due to unexpected rendering, potentially without an easy way to understand what went wrong here. To help out developers,
+			// we are explicitly checking for this scenario here and throw a helpful error message so that they can quickly fix their app.
+			if (newContent is Windows.UI.Xaml.Controls.NavigationViewItemBase)
```

Comment:
just make sure the type here doesn't get converted by the Uno.WinUI transformation :p

Reply:
How do I do that though 😅

Reply:
Worked around :-)

Reply:
in other contexts the 'fix'seems to be to break the string-matching without changing the semantics, so something like 

```csharp
Windows/*Don't convert this*/.UI.Xaml.Controls.NavigationViewItemBase
```